src/Microsoft.VisualStudio.Editors/PropPages/CodeAnalysisPropPage.vb

```diff
@@ -0,0 +1,246 @@
+Imports Microsoft.VisualStudio.Shell.Interop
+Imports NuGet.VisualStudio
+
+Namespace Microsoft.VisualStudio.Editors.PropertyPages
+
+    Friend NotInheritable Class CodeAnalysisPropPage
+        Inherits PropPageUserControlBase
+
+        Private Const NugetFeed As String = "https://packages.nuget.org/api/v2"
+        Private Const FxCopAnalyzersPackageId As String = "Microsoft.CodeAnalysis.FxCopAnalyzers"
+
+        Private Shared ReadOnly s_latestStableVersion As Version = New Version(2, 9, 3)
+        Private Shared ReadOnly s_childPackageIds As HashSet(Of String) = New HashSet(Of String)(
+            {"Microsoft.CodeQuality.Analyzers", "Microsoft.NetCore.Analyzers", "Microsoft.NetFramework.Analyzers"},
+            StringComparer.OrdinalIgnoreCase)
+
+        Private _packageInstallerServices As IVsPackageInstallerServices
+        Private _packageInstaller As IVsPackageInstaller2
+        Private _packageUninstaller As IVsPackageUninstaller
+        Private _packageRestorer As IVsPackageRestorer
+        Private _threadedWaitDialogFactory As IVsThreadedWaitDialogFactory
+        Private _nugetPackage As IVsPackage
+        Private _installedFxCopAnalyzersVersionString As String
+
+        Public Sub New()
+            MyBase.New()
+
+            InitializeComponent()
+            RefreshInstallFxCopAnalyzersButtons(reset:=True)
+
+            'Opt out of page scaling since we're using AutoScaleMode
+            PageRequiresScaling = False
+
+            'Add any initialization after the InitializeComponent() call
+            AddChangeHandlers()
+        End Sub
+
+        Protected Overrides Sub PostInitPage()
+            MyBase.PostInitPage()
+
+            Dim componentModel = CType(ServiceProvider.GetService(GetType(SComponentModel)), IComponentModel)
+            _packageInstallerServices = componentModel.GetExtensions(Of IVsPackageInstallerServices)().SingleOrDefault()
+            _packageInstaller = componentModel.GetService(Of IVsPackageInstaller2)()
+            _packageUninstaller = componentModel.GetService(Of IVsPackageUninstaller)()
+            _packageRestorer = componentModel.GetService(Of IVsPackageRestorer)()
+            _threadedWaitDialogFactory = CType(ServiceProvider.GetService(GetType(SVsThreadedWaitDialogFactory)), IVsThreadedWaitDialogFactory)
+
+            Dim shell = CType(ServiceProvider.GetService(GetType(SVsShell)), IVsShell)
+            Dim nugetGuid = New Guid("5fcc8577-4feb-4d04-ad72-d6c629b083cc")
+            shell?.LoadPackage(nugetGuid, _nugetPackage)
+
+            RefreshInstalledFxCopAnalyzersVersionAndButtons()
+        End Sub
+
+        Protected Overrides ReadOnly Property ControlData() As PropertyControlData()
+            Get
+                If m_ControlData Is Nothing Then
+                    m_ControlData = New PropertyControlData() {
+                        New PropertyControlData(1, "RunRoslynAnalyzersDuringBuild", RunRoslynAnalyzersDuringBuild, ControlDataFlags.None),
```

Comment:
Is there a PR coming for the legacy project system to support these properties, or was it intended that this property page only apply to SDK-style projects?

Reply:
Yes, hopefully soon. It turned out to be lot more non-trivial in the legacy project system then I anticipated (I looked at the logic for existing properties on Code Analysis property page). Probably there is an easier way that I am just not aware about...